### PR TITLE
Estimator

### PR DESCRIPTION
**Pull request by ortyomka**
Signed-off-by: ortyomka <iurin.art@gmail.com>

### CODE REVIEW

File: core/miner/main/main.cpp [hunk 1]
```diff
@@ -425,7 +427,11 @@ namespace fc {
     IoThread io_thread2;
     OUTCOME_TRY(wscheduler,
                 sector_storage::SchedulerImpl::newScheduler(
-                    io_thread2.io, prefixed("scheduler_works/")));
+                    io_thread2.io,
+                    prefixed("scheduler_works/"),
+                    // TODO(ortyomka): make param size
```
Comment (Alexey-N-Chernyshov) [inline core/miner/main/main.cpp:432]: Please, add to miner config as "estimator window" or something like that.

File: core/sector_storage/impl/worker_estimator_impl.hpp [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        if (data_.empty()) {
+          return boost::none;
+        }
+
+        if (dirty_flag_) {
+          auto size = data_.size();
+
+          uint64_t sum = 0;
+          for (size_t i = 0; i < size; i++) {
+            sum += data_.at(i) * (size - i);
```
Comment (Alexey-N-Chernyshov) [inline core/sector_storage/impl/worker_estimator_impl.hpp:66]:
```suggestion
            sum += data_.at(i) * (i + 1);
```
Most recent must have a highest weight.

Comment (turuslan) [inline core/sector_storage/impl/worker_estimator_impl.hpp:66]: may simplify
```suggestion
          auto weight{size};
          for (const auto &value : data_) {
            sum += value * weight;
            --weight;
```

File: core/sector_storage/impl/worker_estimator_impl.hpp [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+        return average_;
+      }
+
+     private:
+      std::deque<uint64_t> data_;
+      uint64_t count_calls_;
+
+      mutable bool dirty_flag_ = false;
+      mutable double average_;
```
Comment (turuslan) [inline core/sector_storage/impl/worker_estimator_impl.hpp:82]: can combine with `optional`?

**Review by turuslan — APPROVED** [review]

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Pull request merged**